### PR TITLE
fix blob api usage

### PR DESCRIPTION
**Pull request by yvesgoeleven**
Fixes the asp sample, so that api works for netcore

### CODE REVIEW

**Comment by yvesgoeleven** [comment]
@Particular/docs-maintainers this fixes the master build

File: samples/azure/storage-persistence/ASP_2/StorageReader/AzureHelper.cs [hunk 1]
```diff
@@ -31,45 +32,61 @@ public Task WriteOutData()
...
+        BlobContinuationToken token = null;
+        do
         {
-            var name = blob.Uri.AbsolutePath.Split('/').Last();
-            Debug.WriteLine($"  Blob:= {name}");
-            var blockBlobReference = container.GetBlockBlobReference(name);
-            var text = await blockBlobReference.DownloadTextAsync()
-                .ConfigureAwait(false);
-            Debug.WriteLine($"    {text}");
+            var segment = await container.ListBlobsSegmentedAsync(token);
```
Comment (danielmarbach) [inline samples/azure/storage-persistence/ASP_2/StorageReader/AzureHelper.cs]: I think this requires `ConfigureAwait(false)` we apply that to the doco snippets

File: samples/azure/storage-persistence/ASP_2/StorageReader/AzureHelper.cs [hunk 2]
```diff
@@ -4,21 +4,22 @@
...
+    public async Task WriteOutData()
     {
         #region UsingHelpers
 
-        WriteOutTable("OrderSagaData", false);
-        WriteOutTable("Subscription", true);
-        WriteOutTable("TimeoutDataTableName", false);
-        WriteOutTable("TimeoutManagerDataTable", false);
-        return WriteOutBlobContainer("timeoutstate");
+        await WriteOutTable("OrderSagaData", false);
```
Comment (danielmarbach) [inline samples/azure/storage-persistence/ASP_2/StorageReader/AzureHelper.cs]: I think this requires `ConfigureAwait(false)` we apply that to the doco snippets

File: samples/azure/storage-persistence/ASP_2/StorageReader/AzureHelper.cs [hunk 3]
```diff
@@ -31,45 +32,61 @@ public Task WriteOutData()
...
-            Debug.WriteLine($"    RowKey:= {entity.RowKey}");
-            if (decodeRowKey)
-            {
-                var decodedRowKey = entity.RowKey.DecodeFromKey();
-                Debug.WriteLine($"    DecodedRowKey:= {decodedRowKey}");
-            }
-            foreach (var property in entity.Properties)
+        TableContinuationToken token = null;
+        do {
+            var result = await table.ExecuteQuerySegmentedAsync(new TableQuery(), token);
```
Comment (danielmarbach) [inline samples/azure/storage-persistence/ASP_2/StorageReader/AzureHelper.cs]: I think this requires `ConfigureAwait(false)` we apply that to the doco snippets

File: samples/azure/storage-persistence/ASP_2/StorageReader/AzureHelper.cs [hunk 4]
```diff
@@ -31,45 +32,61 @@ public Task WriteOutData()
...
-            Debug.WriteLine($"  PartitionKey:= {entity.PartitionKey}");
-            Debug.WriteLine($"    RowKey:= {entity.RowKey}");
-            if (decodeRowKey)
-            {
-                var decodedRowKey = entity.RowKey.DecodeFromKey();
-                Debug.WriteLine($"    DecodedRowKey:= {decodedRowKey}");
-            }
-            foreach (var property in entity.Properties)
+        TableContinuationToken token = null;
+        do {
```
Comment (danielmarbach) [inline samples/azure/storage-persistence/ASP_2/StorageReader/AzureHelper.cs]: Not sure if this formatting is correct. Can you reformat this?

File: samples/azure/storage-persistence/ASP_2/StorageReader/AzureHelper.cs (hunk 2 above)
Comment (yvesgoeleven) [reply samples/azure/storage-persistence/ASP_2/StorageReader/AzureHelper.cs]: addressed

File: samples/azure/storage-persistence/ASP_2/StorageReader/AzureHelper.cs (hunk 1 above)
Comment (yvesgoeleven) [reply samples/azure/storage-persistence/ASP_2/StorageReader/AzureHelper.cs]: addressed

File: samples/azure/storage-persistence/ASP_2/StorageReader/AzureHelper.cs (hunk 4 above)
Comment (yvesgoeleven) [reply samples/azure/storage-persistence/ASP_2/StorageReader/AzureHelper.cs]: done

File: samples/azure/storage-persistence/ASP_2/StorageReader/AzureHelper.cs (hunk 3 above)
Comment (yvesgoeleven) [reply samples/azure/storage-persistence/ASP_2/StorageReader/AzureHelper.cs]: addressed

**Pull request merged**